testing/e2e/skr-tester/pkg/command/update.go

```diff
@@ -0,0 +1,157 @@
+package command
+
+import (
+	"errors"
+	"fmt"
+	"os/exec"
+	broker "skr-tester/pkg/broker"
+	"skr-tester/pkg/logger"
+	"strings"
+
+	"github.com/spf13/cobra"
+)
+
+type UpdateCommand struct {
+	cobraCmd          *cobra.Command
+	log               logger.Logger
+	instanceID        string
+	planID            string
+	updateMachineType bool
+	// TODO
+	updateOIDC bool
+}
+
+func NewUpdateCommand() *cobra.Command {
+	cmd := UpdateCommand{}
+	cobraCmd := &cobra.Command{
+		Use:     "update",
+		Aliases: []string{"u"},
+		Short:   "Update the instnace",
+		Long:    "Update the instnace",
+		Example: "skr-tester update -i instanceID -p planID --updateMachineType                            Update the instance with new machineType.",
+
+		PreRunE: func(_ *cobra.Command, _ []string) error { return cmd.Validate() },
+		RunE:    func(_ *cobra.Command, _ []string) error { return cmd.Run() },
+	}
+	cmd.cobraCmd = cobraCmd
+
+	cobraCmd.Flags().StringVarP(&cmd.instanceID, "instanceID", "i", "", "InstanceID of the specific instance.")
+	cobraCmd.Flags().StringVarP(&cmd.planID, "planID", "p", "", "PlanID of the specific instance.")
+	cobraCmd.Flags().BoolVarP(&cmd.updateMachineType, "updateMachineType", "m", false, "Should update machineType.")
+
+	return cobraCmd
+}
+
+func (cmd *UpdateCommand) Run() error {
+	cmd.log = logger.New()
+	brokerClient := broker.NewBrokerClient(broker.NewBrokerConfig())
+	catalog, err := brokerClient.GetCatalog()
+	if err != nil {
+		return fmt.Errorf("failed to get catalog: %v", err)
+	}
+	services, ok := catalog["services"].([]interface{})
+	if !ok {
+		return errors.New("services field not found or invalid in catalog")
+	}
+	for _, service := range services {
+		serviceMap, ok := service.(map[string]interface{})
+		if !ok {
+			return errors.New("service is not a map[string]interface{}")
+		}
+		if cmd.updateMachineType {
+			currentMachineType, err := getCurrentMachineType(cmd.instanceID)
+			if err != nil {
+				return fmt.Errorf("failed to get current machine type: %v", err)
+			}
+			fmt.Printf("Current machine type: %s\n", *currentMachineType)
+			if serviceMap["id"] != "47c9dcbf-ff30-448e-ab36-d3bad66ba281" {
```

Comment:
What do you think about checking the service as a first condition for the code execution? We are only interested in updating the Kyma instance, so we could get the service by ID from `services` slice and work with that instead iterating through the slice. You can add a const to hold `47c9dcbf-ff30-448e-ab36-d3bad66ba281` value. Looks like the ID is interface{} type when you read it from the map (type of map[string]interface{}).

Reply:
Moved the condition to the top of the services loop, and now I use the const form broker package with the serviceID.